diff -Nur opensim-/OpenSim/Framework/Servers/HttpServer/BaseHttpServer.cs opensim/OpenSim/Framework/Servers/HttpServer/BaseHttpServer.cs
--- opensim-/OpenSim/Framework/Servers/HttpServer/BaseHttpServer.cs	2011-05-05 18:03:50.000000000 +0900
+++ opensim/OpenSim/Framework/Servers/HttpServer/BaseHttpServer.cs	2011-05-05 18:03:05.000000000 +0900
@@ -1702,6 +1702,12 @@
                 {
                     //m_httpListener.Prefixes.Add("https://+:" + (m_sslport) + "/");
                     //m_httpListener.Prefixes.Add("http://+:" + m_port + "/");
+					System.Security.Cryptography.X509Certificates.X509Certificate2 cert =
+					         new System.Security.Cryptography.X509Certificates.X509Certificate2("SineWaveCert.pfx", "123");
+					m_httpListener2 = CoolHTTPListener.Create(IPAddress.Any, (int)m_port, cert);
+					m_httpListener2.ExceptionThrown += httpServerException;
+					m_httpListener2.LogWriter = httpserverlog;
+					//m_httpListener2.DisconnectHandler = httpServerDisconnectMonitor; 
                 }
 
                 m_httpListener2.RequestReceived += OnRequest;
